Mimir.Worker/Handler/LegacyAccountHandler.cs

```diff
@@ -45,12 +53,86 @@ private State ConvertToState(Address address, IValue rawState)
+            case var addr
+                when derivedAddresses.TryGetValue(
+                    "ArenaScoreAddresses",
+                    out var arenaScoreAddresses
+                ) && arenaScoreAddresses.Contains(addr):
+                return ConvertToArenaScoreState(addr, context.RawState);
+            case var addr
+                when derivedAddresses.TryGetValue("ArenaInfoAddresses", out var arenaInfoAddresses)
+                    && arenaInfoAddresses.Contains(addr):
+                return ConvertToArenaInformationState(addr, context.RawState);
             default:
                 throw new InvalidOperationException("The provided address has not been handled.");
         }
     }
 
+    private Dictionary<string, List<Address>> GetDerivedFromAvatarsAddresses(List<Transaction> txs)
+    {
+        var addresses = new Dictionary<string, List<Address>>
+        {
+            { "ArenaScoreAddresses", new List<Address>() },
+            { "ArenaInfoAddresses", new List<Address>() }
+        };
+
+        foreach (var tx in txs)
+        {
+            var action = (Dictionary)tx.Actions[0];
+            var actionType = (Text)action["type_id"];
+            var actionValues = action["values"];
+
+            if (Regex.IsMatch(actionType, "^battle_arena[0-9]*$"))
+            {
+                var avatarAddress = new Address(((Dictionary)actionValues)["maa"]);
+
+                for (int championshipId = 0; championshipId <= 50; championshipId++)
+                {
+                    for (int roundId = 0; roundId <= 7; roundId++)
+                    {
+                        addresses["ArenaScoreAddresses"]
+                            .Add(ArenaScore.DeriveAddress(avatarAddress, championshipId, roundId));
+                        addresses["ArenaInfoAddresses"]
+                            .Add(
+                                ArenaInformation.DeriveAddress(
+                                    avatarAddress,
+                                    championshipId,
+                                    roundId
+                                )
+                            );
+                    }
+                }
+            }
+        }
+
+        return addresses;
+    }
+
+    private ArenaInformationState ConvertToArenaInformationState(Address address, IValue state)
+    {
+        if (state is List list)
+        {
+            return new ArenaInformationState(address, new ArenaInformation(list));
+        }
+        else
+        {
+            throw new ArgumentException("Invalid state type. Expected List.", nameof(state));
+        }
+    }
+
+    private ArenaScoreState ConvertToArenaScoreState(Address address, IValue state)
+    {
+        if (state is List list)
+        {
+            return new ArenaScoreState(address, new ArenaScore(list));
+        }
+        else
+        {
+            throw new ArgumentException("Invalid state type. Expected List.", nameof(state));
+        }
+    }
```

Comment:
It seems better to know its type too.
```suggestion
    private ArenaInformationState ConvertToArenaInformationState(Address address, IValue state)
    {
        if (state is List list)
        {
            return new ArenaInformationState(address, new ArenaInformation(list));
        }
        else
        {
            throw new ArgumentException($"Invalid state type. Expected List. Actual: {state.GetType().Name}", nameof(state));
        }
    }

    private ArenaScoreState ConvertToArenaScoreState(Address address, IValue state)
    {
        if (state is List list)
        {
            return new ArenaScoreState(address, new ArenaScore(list));
        }
        else
        {
            throw new ArgumentException($"Invalid state type. Expected List.  Actual: {state.GetType().Name}", nameof(state));
        }
    }
```